common/generated/intel_acm.cpp

```diff
@@ -0,0 +1,92 @@
+// This is a generated file! Please edit source .ksy file and use kaitai-struct-compiler to rebuild
+
+#include "intel_acm.h"
+#include "../kaitai/exceptions.h"
+
+intel_acm_t::intel_acm_t(kaitai::kstream* p__io, kaitai::kstruct* p__parent, intel_acm_t* p__root) : kaitai::kstruct(p__io) {
+    m__parent = p__parent;
+    m__root = this; (void)p__root;
+    m_header = 0;
+
+    try {
+        _read();
+    } catch(...) {
+        _clean_up();
+        throw;
+    }
+}
+
+void intel_acm_t::_read() {
+    m_header = new header_t(m__io, this, m__root);
+    m_body = m__io->read_bytes((4 * ((header()->module_size() - header()->header_size()) - header()->scratch_space_size())));
+}
+
+intel_acm_t::~intel_acm_t() {
+    _clean_up();
+}
+
+void intel_acm_t::_clean_up() {
+    if (m_header) {
+        delete m_header; m_header = 0;
+    }
+}
+
+intel_acm_t::header_t::header_t(kaitai::kstream* p__io, intel_acm_t* p__parent, intel_acm_t* p__root) : kaitai::kstruct(p__io) {
```

Comment:
```
    - id: rsa_exponent
      type: u4
      if: header_version == 0
```
This is why it can potentially be left uninitialised, and this is the relevant code bit that guarantees that we will not be reading that uninitialised value:
```
        // Add RsaPublicKey
        if (header->_is_null_rsa_exponent() == false) {
            acmInfo += usprintf("ACM RSA Public Key (Exponent: %Xh):", header->rsa_exponent());
        }
        else {
            acmInfo += usprintf("ACM RSA Public Key (Exponent: %Xh):", INTEL_ACM_HARDCODED_RSA_EXPONENT);
        }
```